From 9ce95a93c4facbb9006b1d7305170e378fa1b2c1 Mon Sep 17 00:00:00 2001
From: Punnaiah Choudary Kalluri <punnaiah.choudary.kalluri@xilinx.com>
Date: Tue, 1 Mar 2016 14:46:48 +0530
Subject: [PATCH 677/793] mtd: nand: arasan: Correct the anfc_ecc_matrix for
 4K page size

As per the arasan nand data sheet, number ecc bits should be 24 for
4K page and 1K code word size.

Signed-off-by: Punnaiah Choudary Kalluri <punnaia@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/mtd/nand/arasan_nfc.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/mtd/nand/arasan_nfc.c b/drivers/mtd/nand/arasan_nfc.c
index 94bf3bf..995db07 100644
--- a/drivers/mtd/nand/arasan_nfc.c
+++ b/drivers/mtd/nand/arasan_nfc.c
@@ -133,7 +133,7 @@ static const struct anfc_ecc_matrix ecc_matrix[] = {
 	{4096,	512,	4,	1,	0x34},
 	{4096,	512,	8,	1,	0x68},
 	{4096,	512,	12,	1,	0x9C},
-	{4096,	1024,	4,	1,	0xA8},
+	{4096,	1024,	24,	1,	0xA8},
 	/* 8K byte page */
 	{8192,	512,	1,	0,	0x30},
 	{8192,	512,	4,	1,	0x68},
-- 
1.7.5.4

